.github/workflows/benchmark.yml

```diff
@@ -48,23 +48,13 @@ jobs:
         - 5672:5672
 
     steps:
-    - uses: actions/checkout@v3
-    - name: Set up Python
-      uses: actions/setup-python@v4
+    - uses: actions/checkout@v4
+
+    - name: Install aiida-core
+      uses: ./.github/actions/install-aiida-core
       with:
         python-version: '3.10'
 
-    - name: Upgrade pip
-      run: |
-        pip install --upgrade pip
-        pip --version
-
-    - name: Install python dependencies
-      run: |
-        pip install -r requirements/requirements-py-3.10.txt
-        pip install --no-deps -e .
```

Comment:
Looking at this now, I am not sure why it is first installing from requirements and then doing `--no-deps -e .`. Do you know why? Does installing with `-r` somehow not install the package itself? That cannot be, can it? Or is it because that doesn't support the `-e` flag? But then again, why would we need a linked install for CI?

Just asking to understand if it makes sense that we get rid of this double install approach.